Support Debian 13 / Trixie

Salamandar opened the pull request:
Fix https://github.com/YunoHost/issues/issues/2469

This is a draft PR as Trixie is not released yet.

For Debian Trixie freeze dates, see https://release.debian.org/trixie/freeze_policy.html 

Blockers:

- [x] PHP repository Sury [are waiting for transition freeze](https://github.com/oerdnj/deb.sury.org/issues/2176#issuecomment-2480532004) to start working on a Trixie repository. Most PHP apps are broken until then
- [x] Missing package `python3-lexicon` on Trixie: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1003200
- [x] [Pydantic v2](https://github.com/YunoHost/yunohost/pull/2001) (v1 is not provided by Trixie anymore)
- [x] dovecot configuration
- [ ] postfix service down but isn't ?
- [x] App tests failing
- [x] Pydantic errors in CI
- [x] Migration from Bookworm
  - [ ] `networking.service` failed on VPS after migration?
  - [x] `sssd-*.socket` failed after migration: misconfiguration?
  - [ ] Legacy apt files still present ?

TODO: apt go to stable only

Salamandar committed:
FIXME: remove conflicts with trixie package verisons

Salamandar committed:
nginx now configures 'server_tokens off' in its core config

Salamandar committed:
chore: add python3-pydantic-extra-types dependency

Salamandar committed:
DONOTMERGE: patch because sury doesn't support trixie yet

Salamandar committed:
Replace mysqlshow, mysqldump with mariadb-show, mariadb-dump

See https://mariadb.com/kb/en/changes-and-improvements-in-mariadb-10-5/#binaries-named-mariadb-mysql-symlinked

Axolotle committed:
pydanticv2: Model.dict() -> Model.dump_model()

Axolotle committed:
pydanticv2: Model.__fields__ -> Model.model_fields

Axolotle committed:
pydanticv2: Model.construct() -> Model.model_construct()

Axolotle committed:
pydanticv2: Class Config -> ConfigDict()

Axolotle committed:
pydanticv2: Class Config.schema_extra() -> __get_pydantic_json_schema__()

Axolotle committed:
pydanticv2: validator() -> field_validator()

Axolotle committed:
pydanticv2: root_validator() -> model_validator()

Axolotle committed:
pydanticv2: Field's allow_mutation -> frozen

Axolotle committed:
pydanticv2: regex -> pattern

Axolotle committed:
pydanticv2: fix missing imports

Axolotle committed:
pydanticv2: rework custom validators and extra args

Axolotle committed:
pydanticv2: fix SelectOption and other choices options

Axolotle committed:
pydanticv2: fix TimeOption, was no longer parsing numbers

Axolotle committed:
pydanticv2: fix DateOption, was no longer parsing numbers

Axolotle committed:
pydanticv2: fix PasswordOption, properly handle forbidden fields

Axolotle committed:
pydanticv2: 'ask' as optional

Axolotle committed:
pydanticv2: fix missing imports

Axolotle committed:
pydanticv2: fix pydantic i18n error key might not be available

Axolotle committed:
pydanticv2: fix tests boolean handling in string types

Axolotle committed:
pydanticv2: fix EmailOption tests

Axolotle committed:
pydanticv2: fix UrlOption tests

Axolotle committed:
pydanticv2: fix missing imports

Axolotle committed:
pydanticv2: fix validators return type

Salamandar committed:
Try to run CI on trixie

Salamandar committed:
Use py312 as tox target

Salamandar committed:
Fix lint312 job

Salamandar committed:
Merge pull request #2001 from YunoHost/pydantic-v2

Salamandar committed:
Remove migration to bookworm (it's actually failing and putting weird stuff in output...

Salamandar committed:
Fix migrations after deleting migration to bookworm

Commit:
sury now provides trixie repositories

Axolotle committed:
Merge branch 'dev' into trixie

Axolotle committed:
pydanticv2: Model.dict() -> Model.dump_model()

Salamandar committed:
Replace spwd with direct call to passwd.

spwd was removed in python 3.13

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
tools: passwd now accepts --stdin

Salamandar committed:
Remove non existing udisks2 modules

Salamandar committed:
pydantic: Url can't be imported directly from pydantic.network

Salamandar committed:
Fix _guess_who_started_process when process doesn't have a parent

That happens with 'incus exec yunohost …'

Salamandar committed:
conf-regen: apt: Refactorize with conf files

* Use conf files instead of bash strings
* Replace list files with deb822 source files
* Manage yunohost's sources file and gpg key

Salamandar committed:
conf: Use repo.yunohost.org

Salamandar committed:
conf-regen: apt: Ensure the yunohost repo is not in sources.list

Salamandar committed:
diagnosis: Handle .sources deb822 to find yunohost testing remote

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
small debian/control modification to reduce the diff with dev branch

Salamandar committed:
Remove dependency on dovecot-antispam, not present on trixie and unmaintained

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
Set Python version to 3.13 for Trixie

Thovi98 committed:
increase postgresql version from 15 to 17

alexAubin committed:
Merge pull request #2155 from Thovi98/fix-postgresql-version

[Trixie] Increase postgresql version from 15 to 17

alexAubin committed:
Merge remote-tracking branch 'origin/dev' into trixie

alexAubin committed:
trixie: Fix postgresql version

alexAubin committed:
For some reason dpkg --list | grep doesn't work -______-

alexAubin committed:
apps: Yolodrop support for packaging v1

alexAubin committed:
Merge remote-tracking branch 'origin/dev' into trixie

Salamandar committed:
Merge branch 'dev' into trixie

fflorent committed:
Ensure that ynh_string_random returns a result (#2181)

* test ynh_string_random

* Ensure that the ynh_string_random returns a result

this function can be passed a filter and a length.
But when combined, prior to this fix, we might not
get a result. For example, by running several times the following
command:
`ynh_string_random --length=64 --filter='a-f0-9'`

This is because we used to limit the number of characters output by
/dev/urandom

Now we read as many as random characters as needed to get a random
string given the passed filter and length.

Credits to Salamandar for the fix and to tituspijean for discovering the
issue.

Co-authored-by: Salamandar <6552989+Salamandar@users.noreply.github.com>

---------

Co-authored-by: Salamandar <6552989+Salamandar@users.noreply.github.com>

Salamandar committed:
FIXME: conf_regen/43-dnsmasq: fix do_init_regen

alexAubin committed:
permissions: fix edge case where there's some unknown key in system perm conf, previous code was triggering a 'dictionary changed size during iteration' exception

alexAubin committed:
:art: Format Python code with Ruff, and Bash code with Shfmt

alexAubin committed:
Typo, typing T_T

alexAubin committed:
Zgrompf

alexAubin committed:
:art: Format Python code with Ruff, and Bash code with Shfmt

tituspijean committed:
Update 12.1.24

Salamandar committed:
conf: dovecot: Fix, 2.4.0 expects dovecot_config_version

Salamandar committed:
Update dovecot config for Trixie / dovecot 2.4

Salamandar committed:
Ensure the old yunohost.list file is deleted

Salamandar committed:
diagnosers: dnsrecords: Fix its_important() on missing DNS record

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
changelog: Release yunohost v13.0~alpha1

Salamandar committed:
yarn: Remove Yarn APT repository, warn on apps depending on Yarn via APT

Salamandar committed:
change last version ~ with - for compat ?

Salamandar committed:
change last version - with + for compat ?

Salamandar committed:
debian/changelog: set version to 13.0.0.alpha1

Salamandar committed:
debian/control: Depend on sssd and plugins instead of sudo-ldap, libnss-ldapd, libpam-ldapd

Salamandar committed:
conf: Replace nslcd with sssd

Salamandar committed:
src: permission: Replace nscd call with sss_cache

Salamandar committed:
conf: nsswitch: Replace nslcd with sssd

Salamandar committed:
slapd: stop installing sudo-ldap.conf, it's useless since sssd

Salamandar committed:
hooks: conf_regen: Merge 09-sssd and 46-nsswitch into 09-nss

Salamandar committed:
src: Replace calls to nscd with sss_cache

Salamandar committed:
tests: Fix test_apps

Salamandar committed:
Merge branch 'dev' into HEAD

alexAubin committed:
Update changelog: leading space makes debian sing the song of its people ...

alexAubin committed:
Update changelog: one me is enough

Salamandar committed:
debian/control: dovecot now requires a dependency on dovecot-imapd

Salamandar committed:
conf: dovecot: migrate old %n syntax

Salamandar committed:
conf: postfix: Set compatibility_level

Salamandar committed:
conf: postfix: Sync config file with default Postfix trixie config file

Salamandar commented:
@Axolotle there are some Pydantic related errors, could you please check it out ? Thanks :)

https://gitlab.com/YunoHost/yunohost/-/jobs/11555902636

Salamandar committed:
resources: InstallDir: Stop deleting the install_dir if it exists already

That was a workaround for packaging v1 -> packaging v2 migration.
But this seems dangerous and is not required anymore, let's raise an exception instead.

Salamandar committed:
app: Fix typing for app_upgrade, file can be a dict

Salamandar committed:
tests: apps: break_yo_system is now v2

Salamandar committed:
tests: apps: app_upgrade now requires no_safety_backup=True for dummy v2 apps

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
tests: appurl: Use manifestv2 app for tests

Salamandar committed:
tests: changeurl: use the new v2 app

Salamandar committed:
tests: permissions: Use packaging v2 dummy app

Salamandar committed:
tests: permissions: Use packaging v2 test app

Salamandar committed:
helpers: utils: Fix usage by the core with no YNH_APP_ACTION

Salamandar committed:
helpers: use quotes around variable

Salamandar committed:
pyproject: Add missing pytest markers

Salamandar committed:
tests: backuprestore: Remove usage of legacy app

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
Replace use of deprecated datetime utcnow and utcfromtimestamp

They are replaced with tz-aware counterparts

Salamandar committed:
Run ruff format

Salamandar committed:
utils/form: Replace deprecated use of ast.Str, Num with ast.Constant

Salamandar committed:
Replace OpenSSL with cryptography

OpenSSL x.509 support is deprecated. Remove OpenSSL usage altogether.

Salamandar committed:
Remove dependency on pyOpenSSL

Salamandar committed:
tests: app_config: Fix use of legacy app

Salamandar committed:
certificate: Fix _get_status()

Salamandar committed:
tests: app_config: Fix use of v2 app

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
form: Fix SelectOption with Pydanticv2 that doesn't like empty choices

Salamandar committed:
Fix ruff adding a line break

Salamandar committed:
tests: ruff format

Salamandar committed:
form: Remove legacy support of 'choices' in 'string' type question

Salamandar committed:
tests: questions: Remove test for legacy support of string with choices

Salamandar committed:
pyproject: fix dependencies

Salamandar committed:
dns: Use a class Progress instead of a function with attributes

Salamandar committed:
dyndns: subscribe should always take a domain, remove None as default value

Salamandar committed:
domain: Use typing overload to inform type checker that get_parent_domain_of(return_self=True) always returns str

Salamandar committed:
resources: is_public is packagingv1, we don't need to remove it anymore

Salamandar committed:
conf_regen: nginx: We don't need to remove this legacy file anymore

Salamandar committed:
conf_regen: dovecot: We don't need to remove this legacy directory anymore

Salamandar committed:
tools: migrations: Rename to m[0-9]*_.*

Python really doesn't like files starting with a digit.
Add a 'm' prefix (for "migration") to fix issues with that.

This commit only renames files.
A second commit will fix the rest of the code.

Salamandar committed:
backup: Fix type issues, unimplemented method

Salamandar committed:
backup: Pass filter=data to tarfile.extract()

Behaviour will change with Python 3.14, applying the change now

Salamandar committed:
utils: dns: Replace deprecated call to resolver.query() with resolver.resolve(search=True)

Salamandar committed:
backup: restore actually requires tarfile filter=tar, because filter=data refuses to extract symlinks to absolute paths

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
conf: apt: https for repo.yunohost.org

Salamandar committed:
utils: system: Reimplement /etc/os-release parsing with pure python, add os_release()

Salamandar committed:
hooks: conf-regen: nss: Fix sssd file permission on post_regen

alexAubin committed:
nginx conf: update security conf / ciphers with recommendation from Mozilla for nginx 1.26, openssl 3.5

alexAubin committed:
postfix: update security options according to Mozilla recommendation

alexAubin committed:
dovecot: update security stuff according to mozilla recommentation

alexAubin commented on .gitlab/ci/lint.gitlab-ci.yml:
```diff
@@ -11,27 +11,27 @@ actionsmap:
+    - tox -e py312-lint
 
-invalidcode311:
+invalidcode312:
```
Eeeeh confused at how these can be passing considering its 3.13 in tox.ini ?

Salamandar replied:
You're right, tox is failing now.

alexAubin commented on src/tools.py:
```diff
@@ -800,15 +801,6 @@ def get_matching_migration(target):
 
         raise YunohostValidationError("migrations_no_such_migration", id=target)
 
-    # Dirty hack to mark the bullseye->bookworm as done ...
-    # it may still be marked as 'pending' if for some reason the migration crashed,
-    # but the admins ran 'apt full-upgrade' to manually finish the migration
-    # ... in which case it won't be magically flagged as 'done' until here
-    migrate_to_bookworm = get_matching_migration("migrate_to_bookworm")
-    if migrate_to_bookworm.state == "pending":
-        migrate_to_bookworm.state = "done"
-        _write_migration_state(migrate_to_bookworm.id, "done")
-
```
Not 100% sure but we probably want something similar for the trixie migration

alexAubin commented on conf/sssd/sssd.conf:
```diff
@@ -0,0 +1,29 @@
+chpass_provider = ldap
+sudo_provider = ldap
+
+ldap_uri = ldaps://localhost:636
```
Hmpf i insist that it would really be great it we could make it work without TLS because it's useless overhead for local-loop networking and sooner or later it's gonna create stupid issues like "certificate is not trusted" because of whatever reason, in turn destroying SSH connectivity for admins etc...

Or at least add a comment as to why exactly TLS is needed

Salamandar replied:
> Or at least add a comment as to why exactly TLS is needed

Couldn't manage to make it work without the "s"… So i left it :D

artlog replied:
I could propose this configuration that is working in my trixe dev env and does not rely on any TLS

```
[domain/LDAP]
id_provider = ldap
auth_provider = ldap
chpass_provider = ldap
sudo_provider = ldap

ldap_uri = ldap://localhost:389
ldap_search_base = dc=yunohost,dc=org

ldap_tls_reqcert = never

cache_credentials = True
```

artlog replied:
proposal : https://github.com/YunoHost/yunohost/pull/2295

alexAubin commented on debian/control:
```diff
@@ -44,12 +44,13 @@ Conflicts: iptables-persistent
+#  , slapd (>= 2.6)
+#  , dovecot-core (>= 1:2.4)
+#  , fail2ban (>= 1.1)
+#  , nftables (>= 1.1)
```
```suggestion
  , nginx-extras (>= 1.28)
  , openssl (>= 3.5.4)
  , slapd (>= 2.6.11)
  , dovecot-core (>= 1:2.5)
  , fail2ban (>= 1.2)
  , nftables (>= 1.1.5)
```

alexAubin committed:
Update Conflicts in debian/control

alexAubin commented on debian/control:
```diff
@@ -44,12 +44,12 @@ Conflicts: iptables-persistent
+  , slapd (>= 2.6.11)
+  , dovecot-core (>= 1:2.5)
+  , fail2ban (>= 1.2)
+  , nftables (>= 1.1.5)
```
```suggestion
 , nginx-extras (>= 1.28)
 , openssl (>= 3.5.4)
 , slapd (>= 2.6.11)
 , dovecot-core (>= 1:2.5)
 , fail2ban (>= 1.2)
 , nftables (>= 1.1.5)
```

Salamandar committed:
hooks: postgresql : Use helper to prevent warning messages

Salamandar committed:
utils: Update file_utils for pathlib compat

Salamandar committed:
conf: dovecot: Fix mail_plugins definition in config file

Salamandar committed:
conf: yunohost: services: In Trixie, postfix main runner is not postfix@- anymore

Salamandar committed:
pyproject: Trixie has python 3.13 now!

Salamandar committed:
debian: Trixie has python 3.13 now!

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
maintenance: chmod +x manual scripts

Salamandar committed:
make_changelog.sh: filter 13.x tags

Salamandar committed:
conf: dovecot: Add back the global_script before sieve script

Salamandar committed:
conf: dovecot: Remove legacy, non existing antispam plugin

Salamandar committed:
conf: Rename all jinja templates as *.j2

Salamandar committed:
conf: dovecot: replace sed with jinja expression

Salamandar committed:
conf: postfix: replace sed and file creation with jinja expression

Salamandar committed:
Fix linting: We're now on python 3.13

Salamandar committed:
src: Fix flake8 issues for Python 3.13

Salamandar committed:
Fix mypy issues for python 3.13

Salamandar committed:
pyproject: Add pydantic_extra_types to the list of dependencies that don't provide typing for mypy

Salamandar committed:
conf: nginx: Oops, some files weren't jinja templates

Salamandar committed:
conf: nginx: j2 file was not renamed, oops

Salamandar committed:
conf_regen: postgresql: Ooops, test checking if postgresql was installed was inverted

Salamandar committed:
hooks: conf_regen: nss: Fix post_regen, chmod before restarting the server

Salamandar committed:
certificate: cryptography.serialization.load_pem_private_key expects binary data

alexAubin committed:
debian/control: tweak openssl conflict rule

alexAubin committed:
Re-add changelog for 13.0.0 that disappeared because of some forcepush or something...

alexAubin committed:
Merge remote-tracking branch 'origin/dev' into trixie

alexAubin committed:
Update changelog for 13.0.1

Salamandar committed:
tests: file_utils: Fix tests now that we use pathlib

Salamandar committed:
debian: Add an explicit dependency on sudo: libsss-sudo doesn't actually depend on sudo...

kay0u committed:
sse: Improve heartbeat handling and error logging in sse_stream, 10ms was a bit expensive for no reason

kay0u committed:
log: Update log cache symlink handling to accept configurable age parameter, change since_days_ago default param to 1 month

kay0u committed:
remove useless fields in sssd conf

Salamandar committed:
conf: nginx: Fix http2 setting

Salamandar committed:
conf: nginx: Fix jinja template to remove empty lines

Salamandar committed:
conf-regen: apt: Fix when sources.list does not exist on postinstall

alexAubin committed:
Merge remote-tracking branch 'origin/dev' into trixie

alexAubin committed:
Update changelog for 13.0.3

kay0u committed:
fix debian_version_id, return type to Literal

alexAubin committed:
Merge pull request #2242 from YunoHost/fix-debian-version-id

fix debian_version_id, return type to Literal

Salamandar committed:
conf_regen: yunohost: Simplify systemd daemon-reload logic

Just check if the /etc/systemd/system directory was edited.

Signed-off-by: Félix Piédallu <felix@piedallu.me>

Salamandar committed:
conf, hooks: Replace cron jobs with systemd timers/services

Salamandar committed:
conf_regen: yunohost: Fix file touched in /etc instead of pending_dir

Salamandar committed:
conf_regen: dnsmasq: Replace hand-written file with jinja template

Salamandar committed:
conf_regen: yunohost: small reorg

Salamandar committed:
system: ignore mypy issue caused by Literal int return value

Salamandar committed:
app_catalog: ignore crappy mypy error

Salamandar committed:
dns: formatting

Salamandar committed:
utils: system: formatting

Salamandar committed:
helpers: apt: sury php repository was renamed. Also, use if then instead of ||

Salamandar committed:
conf: apt: Install repo GPG keys in /usr/share/keyrings

Salamandar committed:
conf_regen: apt: gpg dearmor directly to file without pipe

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
helpers: apt: Use deb822, install keys in /usr/share/keyrings, rename configuration files to prevent conflicts

Salamandar committed:
tools: Use pkgutil to iterate over and import migrations

This is more solid than iterating on the files

Salamandar committed:
tools: migrations modules are now prefixed with 'm' but ids are unchanged

tituspijean committed:
fix: expired influxdata repo key (#2253)

tituspijean committed:
Update changelog for 13.0.4

Salamandar committed:
hook: typing

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
dyndns: typing

Salamandar committed:
utils.misc: Add send_admin_email, use it in certificate and diagnosis.

Salamandar committed:
certificate: Move tail to utils.file_utils

Salamandar committed:
dyndns: Add an email option to the update operation to send failures by mail to admin

ewilly committed:
Fix #2751 : Update udisks2_interfaces.py

ewilly committed:
Update regenconf.py

Salamandar committed:
Merge pull request #2212 from YunoHost/rename_migrations

tools: migrations: Rename to m[0-9]*_.*

Salamandar committed:
diagnosis, certificate: Fix error email formatting

Salamandar committed:
firewall: upnp: Use Systemd timer instead of a cron file for updates

Salamandar committed:
conf_regen: yunohost: Cleanup cron-related code

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
bin/*: Apply better formatting (run ruff strict)

Salamandar committed:
doc, maintenance: Modernizing python

Salamandar committed:
Add cast to str to silent linter

Salamandar committed:
doc: generate_zsh_completion: Fix linter

Salamandar committed:
[FIXME] conf: apt: Enable yunohost testing until trixie is released as stable

Salamandar committed:
yunohost-renew-certs: Remove some systemd protections

These prevent postmap to be run

Salamandar committed:
sssd: set enumerate=true for ldap

this is required for getent passwd to list ldap users

Salamandar committed:
dyndns: Add a todo about the dynette changes

Salamandar committed:
Merge branch 'dev' into trixie

Salamandar committed:
__init__: Use Pathlib instead of os

Salamandar committed:
Revert "sssd: set enumerate=true for ldap"

This reverts commit 4956d3bb71930b1ca99ad485df6398c6bd662ab4.

Commit:
trixie: configure sssd_ldap to use ldap:// on localhost without any TLS

remove anything about tls, switch from ldaps:// to ldap://

still some extra info about removed fields that were not relevant anyway :

remove ldap_tls_cacert
/etc/yunohost/certs/yunohost.org/crt.pem is a self signed certificate
it is for yunohost.org that won't match
it is not a CA ( CA:FALSE )
found information indicating self-signed certifcates can't be verified with ssdd anyway
see ( https://docs.pagure.org/sssd.sssd/users/faq.html )

remove ldap_tls_key
this would not be required even with tls
it is for client authentication and we don't provide ldap_tls_cert
afaik ynh ldap server is not configured to accept client certificates

to get it working correctly in tls :
a real certificate with a trusted ca should be used
ldap_tls_reqcert = demand should be used with ldaps://<domain.tld>:636 url

Signed-off-by: philippe lhardy <philippe.lhardy@astrolabe.coop>

Salamandar committed:
helpers: mongodb: Support trixie

Salamandar committed:
Merge remote-tracking branch 'origin/dev' into trixie

Salamandar committed:
Update changelog for 13.0.5

ajoe committed:
configpanel: fix broken OptionType comparisons (is -> ==)

With `use_enum_values = True` on BaseOption, pydantic stores `option.type`
as the plain string value, so identity comparisons against OptionType enum
members are always False. Use equality instead, which works whether the
value is a str or an OptionType member (str-based Enum).

Most visible consequence: any app config panel containing a `file`
question whose bound file does not exist yet fails to load with
"Config panel question 'X' should be initialized with a value during
install or upgrade." The password comparison also silently stopped
masking password values in `config get` output.

Fixes YunoHost/issues#2753

zamentur committed:
Merge branch 'dev' into trixie

zamentur committed:
[fix] option.type is no a string
